controller/RestTestCenterUsers.php

```diff
@@ -0,0 +1,251 @@
+            $user = $this->getUserFromRequest();
+            $role = $this->getRoleFromRequest();
+
+            $this->returnJson([
+                'success' => $this->getService()->assignUser($testCenter, $user, $role)
+            ]);
+        } catch (\common_Exception $e) {
+            return $this->returnFailure($e);
+        }
+    }
+
+    /**
+     * @OA\Delete(
+     *     path="/taoTestCenter/api/testCenterUsers",
+     *     tags={"testCenter"},
+     *     summary="Remove user from the test center",
+     *     description="Remove user from the test center",
+     *     @OA\Parameter(
+     *       name="testCenter",
+     *       in="path",
+     *       description="The test center id",
+     *       required=true,
+     *       @OA\Schema(
+     *           type="string"
+     *       )
+     *     ),
+     *     @OA\Parameter(
+     *       name="user",
+     *       in="path",
+     *       description="User id",
+     *       required=true,
+     *       @OA\Schema(
+     *           type="string"
+     *       )
+     *     ),
+     *     @OA\Parameter(
+     *       name="role",
+     *       in="path",
+     *       description="Role id",
+     *       required=true,
+     *       @OA\Schema(
+     *           enum={"http://www.tao.lu/Ontologies/TAOProctor.rdf#TestCenterAdministratorRole", "http://www.tao.lu/Ontologies/TAOProctor.rdf#ProctorRole"},
+     *           type="string"
+     *       )
+     *     ),
+     *     @OA\Response(
+     *         response="200",
+     *         description="Remove user from the test center",
+     *         @OA\MediaType(
+     *             mediaType="application/json",
+     *             @OA\Schema(
+     *                 type="object",
+     *                 @OA\Property(
+     *                     property="success",
+     *                     type="boolean",
+     *                     description="`false` on failure, `true` on success",
+     *                 ),
+     *                 example={
+     *                     "success": true
+     *                 }
+     *             ),
+     *         ),
+     *     ),
+     * )
+     */
+    public function delete()
+    {
+        try {
+            $testCenter = $this->getTCFromRequest();
+            $user = $this->getUserFromRequest();
+            $role = $this->getRoleFromRequest();
+
+            $this->returnJson([
+                'success' => $this->getService()->unassignUser($testCenter, $user, $role)
+            ]);
+        } catch (\common_Exception $e) {
+            return $this->returnFailure($e);
+        }
+    }
+
+    /**
+     * TODO:
+     */
+    public function get()
+    {
+        //todo: implement retrieving test center users
+    }
+
+    /**
+     * @return \oat\oatbox\user\User
+     * @throws \common_exception_BadRequest
+     * @throws \common_exception_Error
+     * @throws \common_exception_MissingParameter
+     */
+    private function getUserFromRequest()
```

Comment:
The `getUserById()` could return `null` as well so the `@return` should be extended with it. And in that case in L129 the `assignUser()` and in L198 the `unassignUser()` could throw RuntimeException which is not inherited from the `common_Exception`.

Reply:
`getUserById()` cannot return null because id given from request parameter cannot be instance of `core_kernel_classes_Resource` or `core_kernel_users_GenerisUser`;

Reply:
It will return null if the `getUserById()` receives a non-string parameter. So in that case it won't able to parse it a `Resource` or `GenerisUser` and will return `null`. I tested it and fortunately it won't happen, as the parameters from the requests are get parsed always as string, so no action required.